src/tests/functional/shared_test_classes/src/single_op/memory.cpp

```diff
@@ -0,0 +1,177 @@
+    } else {
+        assign = std::make_shared<ov::op::v6::Assign>(add, variable);
+    }
+
+    auto res = std::make_shared<ov::op::v0::Result>(add);
+    function = std::make_shared<ov::Model>(ResultVector{res}, SinkVector{assign}, param, "TestMemory");
+}
+
+void MemoryLayerTest::CreateTIFunc(ov::element::Type model_type, ov::Shape input_shape) {
+    auto param = std::make_shared<ov::op::v0::Parameter>(model_type, ov::Shape(input_shape));
+
+    std::vector<std::vector<size_t>> shape = {{static_cast<size_t>(iteration_count), 1}};
+    auto iter_count = std::make_shared<ov::op::v0::Parameter>(model_type, ov::Shape{static_cast<size_t>(iteration_count), 1});
+
+    // Body
+    auto X = std::make_shared<ov::op::v0::Parameter>(model_type, ov::Shape(input_shape));
+    auto Y = std::make_shared<ov::op::v0::Parameter>(model_type, ov::Shape(input_shape));
+    auto Iter = std::make_shared<ov::op::v0::Parameter>(model_type, ov::Shape{1, 1});
+    auto add = std::make_shared<ov::op::v1::Add>(X, Y);
+    auto res = std::make_shared<ov::op::v0::Result>(add);
+    auto Iter_res = std::make_shared<ov::op::v0::Result>(Iter);
+    auto body = std::make_shared<ov::Model>(OutputVector{res, Iter_res}, ParameterVector {X, Y, Iter});
+
+    // TI construction
+    auto tensor_iterator = std::make_shared<ov::op::v0::TensorIterator>();
+    tensor_iterator->set_body(body);
+
+    tensor_iterator->set_merged_input(X, param, res);
+    tensor_iterator->set_invariant_input(Y, param);
+    tensor_iterator->set_sliced_input(Iter, iter_count, 0, 1, 1, -1, 0);
+
+    auto output = tensor_iterator->get_iter_value(res, -1);
+    auto output_iter = tensor_iterator->get_concatenated_slices(Iter_res, 0, 1, 1, -1, 0);
+    function = std::make_shared<ov::Model>(OutputVector{output, output_iter},
+                                           ParameterVector{param, iter_count},
+                                           "PureTI");
+}
+
+void MemoryLayerTest::ApplyLowLatency(ngraph::helpers::MemoryTransformation transformation) {
+    if (transformation == ngraph::helpers::MemoryTransformation::LOW_LATENCY_V2) {
+        function->validate_nodes_and_infer_types();
+        pass::Manager manager;
+        manager.register_pass<pass::LowLatency2>();
+        manager.run_passes(function);
+    } else if (transformation == ngraph::helpers::MemoryTransformation::LOW_LATENCY_V2_ORIGINAL_INIT) {
+        function->validate_nodes_and_infer_types();
+        pass::Manager manager;
+        manager.register_pass<pass::LowLatency2>(false);
+        manager.run_passes(function);
+    }
+}
+
+void MemoryLayerTest::infer() {
+    inferRequest = compiledModel.create_infer_request();
+    for (size_t iter = 0; iter <= iteration_count; iter++) {
+        for (const auto& input : inputs) {
+            inferRequest.set_tensor(input.first, input.second);
+        }
+        inferRequest.infer();
+    }
+}
+
+std::vector<ov::Tensor> MemoryLayerTest::calculate_refs() {
```

Comment:
As I see, the function is a full copy of `SubgraphBaseTest::calculate_refs()`. Please, try to remove it and use basic implementation

Reply:
Unfortunately not a full copy.
It has additional loop over infer() call
`for (size_t iter = 0; iter <= iteration_count; iter++)`

Reply:
@olpipi you're right, I though I removed this comment before submitting 😄